torch/serialization.py

```diff
@@ -753,13 +756,13 @@ def _open_file_like(name_or_buffer, mode):
             raise RuntimeError(f"Expected 'r' or 'w' in mode but got {mode}")
 
 
-class _open_zipfile_reader(_opener):
-    def __init__(self, name_or_buffer) -> None:
+class _open_zipfile_reader(_opener[torch._C.PyTorchFileReader]):
+    def __init__(self, name_or_buffer: Union[str, IO[bytes]]) -> None:
         super().__init__(torch._C.PyTorchFileReader(name_or_buffer))
 
 
-class _open_zipfile_writer_file(_opener):
-    def __init__(self, name) -> None:
+class _open_zipfile_writer_file(_opener[torch._C.PyTorchFileWriter]):
+    def __init__(self, name: str) -> None:
         self.file_stream = None
         self.name = str(name)
```

Comment:
str cast is redundant now ( assuming appropriate type checking?)

Reply:
I removed the cast